src/helpers/checks.jl

```diff
@@ -48,57 +124,288 @@ function check_differential(
+* `slope_tol`         – (`0.1`) tolerance for the slope (global) of the approximation
+* `throw_error`       - (`false`) throw an error message if the gradient is wrong
+Note that `throw_error` disables returning the plot, so better use `io=stdout` if you would like to see the message together with the plot.
+"""
+function check_Hessian(
+    M::AbstractManifold,
+    f,
+    grad_f,
+    Hess_f,
+    p=rand(M),
+    X=rand(M; vector_at=p),
+    Y=rand(M; vector_at=p);
+    a=randn(),
+    b=randn(),
+    check_grad=true,
+    check_vector=true,
+    check_symmetry=true,
+    check_linearity=true,
+    exactness_tol=1e-12,
+    io::Union{IO,Nothing}=nothing,
+    gradient=grad_f(M, p),
+    Hessian=Hess_f(M, p, X),
+    limits=(-8.0, 0.0),
+    mode::Symbol=:Default,
+    N=101,
+    log_range=range(limits[1], limits[2]; length=N),
+    plot=false,
+    retraction_method=default_retraction_method(M, typeof(p)),
+    slope_tol=0.1,
+    throw_error=false,
+    window=nothing,
+    kwargs...,
+)
+    if check_grad
+        if !check_gradient(
+            M, f, grad_f, p, X; gradient=gradient, throw_error=throw_error, io=io, kwargs...
+        )
+            return false
+        end
+    end
+    check_vector && (!is_vector(M, p, Hessian, throw_error) && return false)
+    if check_linearity
+        if !is_Hessian_linear(
+            M, Hess_f, p, X, Y, a, b; throw_error=throw_error, io=io, kwargs...
+        )
+            return false
+        end
+    end
+    if check_symmetry
+        if !is_Hessian_symmetric(
+            M, Hess_f, p, X, Y; throw_error=throw_error, io=io, kwargs...
+        )
+            return false
+        end
+    end
+    if mode === :CriticalPoint # find a critical point and update grad, hess and X
+        p = gradient_descent(M, f, grad_f, p)
+        gradient = grad_f(M, p)
+        X = rand(M; vector_at=p)
+        Hessian = Hess_f(M, p, X)
+    end
+    #
+    # slope check
+    X_n = X ./ norm(M, p, X) # normalize tangent direction
+    Hessian_n = Hessian ./ norm(M, p, X)
+    # function for the directional derivative
+    #
+    T = exp10.(log_range)
+    # points p_i to evaluate our error function at
+    points = map(t -> retract(M, p, X_n, t, retraction_method), T)
+    # F(p_i)
+    costs = [f(M, pi) for pi in points]
+    # linearized
+    linearized = map(
+        t ->
+            f(M, p) +
+            t * real(inner(M, p, gradient, X_n)) +
+            t^2 / 2 * real(inner(M, p, Hessian_n, X_n)),
+        T,
+    )
+    return prepare_check_result(
+        log_range,
+        abs.(costs .- linearized),
+        3.0;
+        exactness_tol=exactness_tol,
+        io=io,
+        name="Hessian",
+        plot=plot,
+        slope_tol=slope_tol,
+        throw_error=throw_error,
+        window=window,
+    )
+end
+
+@doc raw"""
+    is_Hessian_linear(M, Hess_f, p,
+    X=rand(M; vector_at=p), Y=rand(M; vector_at=p), a=randn(), b=randn();
+    throw_error=false, io=nothing, kwargs...
+)
```

Comment:
```suggestion
    is_Hessian_linear(M, Hess_f, p,
        X=rand(M; vector_at=p), Y=rand(M; vector_at=p), a=randn(), b=randn();
        throw_error=false, io=nothing, kwargs...
    )
```
I think it's a bit more readable when indented like this.